quic/s2n-quic-core/src/datagram/default.rs

```diff
@@ -0,0 +1,93 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// s2n-quic's default implementation of the datagram component
+
+use crate::{
+    datagram::unreliable_datagram::{ConnectionInfo, Endpoint, Packet, Receiver, Sender},
+    time::Timestamp,
+};
+use alloc::collections::VecDeque;
+use bytes::Bytes;
+use core::time::Duration;
+
+#[derive(Debug, Default)]
+pub struct Disabled;
+
+// Default capacity is zero right now since the feature is currently disabled.
+const DEFAULT_CAPACITY: usize = 0;
+
+impl Endpoint for Disabled {
+    type Sender = DefaultSender;
+    type Receiver = DisabledReceiver;
+
+    fn create_connection(&mut self, _info: &ConnectionInfo) -> (Self::Sender, Self::Receiver) {
+        let queue = VecDeque::with_capacity(DEFAULT_CAPACITY);
+        (
+            DefaultSender {
+                queue,
+                prioritize_datagrams: true,
+            },
+            DisabledReceiver,
+        )
+    }
+}
+
+pub struct DefaultSender {
+    queue: VecDeque<Datagram>,
+    prioritize_datagrams: bool,
+}
+
+struct Datagram {
+    data: Bytes,
+    creation_time: Timestamp,
+}
+
+// Expiration in ms of datagrams on the queue. If datagrams are older than 3 ms,
+// consider them expired and don't send them.
+const DATAGRAM_EXPIRATION: u64 = 3;
```

Comment:
This should also be configurable. Also, it's better to use `core::time::Duration` for this kind of stuff so the units are carried with its usage rather than inferred through the name.